presto-main/src/main/java/com/facebook/presto/execution/scheduler/nodeSelection/SimpleNodeSelector.java

```diff
@@ -174,29 +206,16 @@ else if (!splitWaitingForAnyNode) {
         return new SplitPlacementResult(blocked, assignment);
     }
 
-    private ResettableRandomizedIterator<InternalNode> getRandomCandidates(int limit, NodeMap nodeMap, List<RemoteTask> existingTasks)
```

Comment:
Frankly, I don't think we need class `RandomNodeSelection` given the abstraction `NodeSelection` is not there anymore. Just keep this function.

Reply:
1. It can separate the logic from SimpleNodeSelector
2. If we revert this change, every time we pick nodes, we have to manually call randomCandidates.reset(); which would be error-prone
3. There is a high chance that we might implement resource-aware nodeSelection , which will be another implementation of NodeSelector